### ISSUE TITLE
Cannot configure custom storage class for backend-redis-storage  , mysql-storage , system-redis-storage

### ISSUE THREAD

**Issue by andreaskaris** [3scale/3scale-operator#336]
TL;DR:
How can I change the storage class for all PVCs created by the apimanager without changing the default storage class. If this is possible, please add documentation for it, if it's not possible, please add the necessary code to make this configurable.

----------------------------------------------------

Cannot configure custom storage class for backend-redis-storage  , mysql-storage , system-redis-storage

When creating the APIManager in step 2.6.1. Deploying the APIManager custom resource of the 3scale installation guide [0], I add the following to the spec:

~~~
  system:
    fileStorage:
      persistentVolumeClaim:
        storageClassName: '<storage class name>'
~~~

For example, if the wildcardDomain is example.com and the storage class name as seen in oc get storageclass is aws-efs create the API manager with the following definition:

~~~
$ cat apimanager.yaml 
apiVersion: apps.3scale.net/v1alpha1
kind: APIManager
metadata:
  name: example-apimanager
spec:
  productVersion: "2.7"
  wildcardDomain: example.com
  resourceRequirementsEnabled: true
  system:
    fileStorage:
      persistentVolumeClaim:
        storageClassName: 'aws-efs'
~~~

However, this will only update the StorageClass for system-storage:
~~~
$ oc get pvc
NAME                    STATUS    VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
backend-redis-storage   Pending                                                                        gp2            14s
mysql-storage           Pending                                                                        gp2            14s
system-redis-storage    Pending                                                                        gp2            14s
system-storage          Bound     pvc-d59027a0-7d8c-4427-b2fe-1800af13aed8   100Mi      RWX            aws-efs        14s
~~~

At the moment, I work around this by changing the default storage class:

Set the current default storage class to false:
~~~
$ oc patch sc gp2 -p '{"metadata":{"annotations": {"storageclass.kubernetes.io/is-default-class":"false"}}}'
~~~

Set the desired storage class as the default class:
~~~
$ oc patch sc aws-efs -p '{"metadata":{"annotations": {"storageclass.kubernetes.io/is-default-class":"true"}}}'
~~~

Verify:
~~~
$ oc get sc
NAME                PROVISIONER             AGE
aws-efs (default)   openshift.org/aws-efs   24h
gp2                 kubernetes.io/aws-ebs   28h
~~~

Create ApiManager and verify:
~~~
$ cat apimanager.yaml 
apiVersion: apps.3scale.net/v1alpha1
kind: APIManager
metadata:
  name: example-apimanager
spec:
  productVersion: "2.7"
  wildcardDomain: example.net
  resourceRequirementsEnabled: true
$ oc create -f apimanager.yaml 
$ oc get pvc
NAME                    STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
backend-redis-storage   Bound    pvc-d8e75d6c-8814-4ac7-9ff8-604ca56116a8   1Gi        RWO            aws-efs        2s
mysql-storage           Bound    pvc-cc8d0f37-bf62-4897-8bf1-95ef77bb1e0c   1Gi        RWO            aws-efs        2s
system-redis-storage    Bound    pvc-bc67130c-4213-4214-bdba-fbd143b03154   1Gi        RWO            aws-efs        2s
system-storage          Bound    pvc-fc4a68e9-2fe8-4c02-b848-2883c4d704e6   100Mi      RWX            aws-efs        2s
~~~

 [0] https://access.redhat.com/documentation/en-us/red_hat_3scale_api_management/2.7/html/installing_3scale/install-threescale-on-openshift-guide#deploying-apimanager-custom-resource

---------------

Thanks,

Andreas

**Comment by eguzki** [comment]
Definitely we will extend our CRD to allow storage class settings per PVC. Openning related Jira issue for our internal tracking

**Comment by eguzki** [comment]
As a side note, `spec.productVersion` field does not exist in APIManager CRD

**Comment by miguelsorianod** [comment]
Configurable storage classes has been added in https://github.com/3scale/3scale-operator/pull/386
